frappe/patches/v12_0/rename_uploaded_files_with_proper_name.py

```diff
@@ -0,0 +1,36 @@
+import frappe
```

Comment:
overall code and be simplified and improved

something like

```py
import frappe

def execute():
	filenames_with_url = frappe.get_all("File", filters={
		"is_folder": 0,
		"file_name": ["like", "%/%"]
	}, fields=['name', 'file_name', 'file_url'])

	for f in filenames_with_url:
		
		# check if file_url exist in the disk... only then proceed
		
		file_name = f.file_name.rsplit('/', 1)[-1]

		if not f.file_url:
			f.file_url = f.file_name

		frappe.db.set_values('File', f.name, {
			"file_name": filename,
			"file_url": f.file_url
		})

	files_without_filenames = frappe.get_all("File", filters={
		"is_folder": 0,
		"file_name": ["is", "not set"],
		"file_url": ["is", "set"]
	}, fields=['file_url', 'name'])

	for f in filenames_with_url:
		file_name = f.file_name.rsplit('/', 1)[-1]
		frappe.db.set_value("File", f.name, "filedname", file_name)

```

Also, make sure actual file exist in the server before renaming